src/Infrastructure/Sofort/Transfer/Request.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Infrastructure\Sofort\Transfer;
+
+use WMDE\Euro\Euro;
+
+class Request {
+
+	/**
+	 * @var Euro
+	 */
+	private $amount;
+	/**
+	 * @var string
+	 */
+	private $currencyCode = '';
+	/**
+	 * @var string
+	 */
+	private $successUrl = '';
+	/**
+	 * @var string
+	 */
+	private $abortUrl = '';
+	/**
+	 * @var string
+	 */
+	private $notificationUrl = '';
+	/**
+	 * @var string[]
+	 */
+	private $reasons = [ '', '' ];
+
+	/**
+	 * @return Euro
+	 */
+	public function getAmount(): Euro {
+		return $this->amount;
+	}
+
+	/**
+	 * @param Euro $amount
+	 */
+	public function setAmount( Euro $amount ): void {
```

Comment:
Me, too. In my opinion there should be either type hints or phpdoc comments (to avoid inconsistency and/or redundance). I would also prefer type hints over phpdoc except there are reasons not to use them.